sys/cbor/cbor.c

```diff
@@ -319,10 +335,14 @@ static size_t decode_int(const cbor_stream_t *s, size_t offset, uint64_t *val)
 
     *val = 0; /* clear val first */
 
+    CBOR_ENSURE_SIZE_READ(s, offset + 1);
```

Comment:
Valid point, but IMO out of scope of this PR. I would propose to open a follow-up PR that adds some assertions to check for null pointers. Okay?